src/libcore/slice.rs

```diff
@@ -1459,12 +1459,30 @@ pub mod bytes {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<A, B> PartialEq<[B]> for [A] where A: PartialEq<B> {
     fn eq(&self, other: &[B]) -> bool {
-        self.len() == other.len() &&
-            order::eq(self.iter(), other.iter())
+        if self.len() != other.len() {
+            return false;
+        }
+
+        for i in 0..self.len() {
+            if !self[i].eq(&other[i]) {
+                return false;
+            }
+        }
+
+        true
     }
     fn ne(&self, other: &[B]) -> bool {
```

Comment:
I know this has already been merged and looks fantastic but I, as a Rust n00b, had a question as to why this isn't just
```
fn ne(&self, other: &[B]) -> bool {
    !self.eq(other);
}
```
Is it a performance issue to avoid the duplication?